### PR TITLE
[14.0][FIX] account_move_line_sale_info: adjust migration script

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
In v14 the account.move has the stock_move_id in standard, so we should make use of it.

@AaronHForgeFlow

### CODE REVIEW

**Review by mariadforgeflow — APPROVED** [review]
LGTM!

File: account_move_line_sale_info/hooks.py [hunk 1]
```diff
@@ -12,7 +12,20 @@ def post_init_hook(cr, registry):
...
+    cr.execute(
+        """
+        UPDATE account_move_line aml SET sale_line_id = sm.sale_line_id
+        FROM account_move_line aml2
+        INNER JOIN account_move am
+        ON am.id = aml2.move_id
+        INNER JOIN stock_move sm ON
+        am.stock_move_id = sm.id
+        WHERE aml.id = aml2.id
+        AND sm.sale_line_id IS NOT NULL;
```
Comment (MiquelRForgeFlow) [inline account_move_line_sale_info/hooks.py:28]:
```suggestion
        AND sm.sale_line_id IS NOT NULL AND aml.sale_line_id IS NULL;
```
same in the migration query

File: account_move_line_sale_info/hooks.py [hunk 2]
```diff
@@ -12,7 +12,20 @@ def post_init_hook(cr, registry):
         FROM account_move_line aml2
         INNER JOIN stock_move sm ON
         aml2.stock_move_id = sm.id
-        WHERE aml.id = aml2.id;
+        WHERE aml.id = aml2.id
+        AND sm.sale_line_id IS NOT NULL;
```
Comment (MiquelRForgeFlow) [inline account_move_line_sale_info/hooks.py:16]:
```suggestion
        AND sm.sale_line_id IS NOT NULL AND aml.sale_line_id IS NULL;
```

File: account_move_line_sale_info/hooks.py (hunk 1 above)
Comment (JordiBForgeFlow) [reply account_move_line_sale_info/hooks.py:28]: @MiquelRForgeFlow Not necessary, if you want to make sure to overwrite

**Review by AaronHForgeFlow — APPROVED** [review]
LGTM, agreed to overwrite the column. Just pensing to fixup the two commits

**Pull request merged**